src/test/bridges/uniswap/UniswapBridgeUnit.t.sol

```diff
@@ -0,0 +1,325 @@
+        uint64 encodedPath = 0x2A8010F003;
+
+        bridge.preApproveTokenPair(LQTY, address(0));
+
+        // Define input and output assets
+        AztecTypes.AztecAsset memory inputAssetA = AztecTypes.AztecAsset({
+            id: 2,
+            erc20Address: LQTY,
+            assetType: AztecTypes.AztecAssetType.ERC20
+        });
+
+        AztecTypes.AztecAsset memory outputAssetA = AztecTypes.AztecAsset({
+            id: 0,
+            erc20Address: address(0),
+            assetType: AztecTypes.AztecAssetType.ETH
+        });
+
+        deal(LQTY, address(bridge), swapAmount);
+
+        (uint256 outputValueA, , ) = bridge.convert(
+            inputAssetA,
+            emptyAsset,
+            outputAssetA,
+            emptyAsset,
+            swapAmount,
+            0,
+            encodedPath,
+            address(0)
+        );
+
+        assertGt(outputValueA, 0);
+    }
+
+    function testSwapHighToLowDecimals(uint80 _swapAmount) public {
+        // Trying to swap anywhere from 0.1 ETH to 10 thousand ETH
+        uint256 swapAmount = bound(_swapAmount, 1e17, 1e22);
+        uint256 tokenInDecimals = 18;
+
+        uint256 quote = QUOTER.quoteExactInput(
+            abi.encodePacked(WETH, uint24(500), USDC, uint24(3000), GUSD),
+            swapAmount
+        );
+        uint256 desiredMinPrice = (quote * 10**tokenInDecimals) / swapAmount;
+
+        uint64 encodedPath = uint64(
+            (bridge.encodeMinPrice(desiredMinPrice) << bridge.SPLIT_PATHS_BIT_LENGTH()) +
+                (bridge.encodeSplitPath(100, 500, USDC, 100, address(0), 3000))
+        );
+
+        bridge.preApproveTokenPair(WETH, GUSD);
+
+        // Define input and output assets
+        AztecTypes.AztecAsset memory inputAssetA = AztecTypes.AztecAsset({
+            id: 2,
+            erc20Address: WETH,
+            assetType: AztecTypes.AztecAssetType.ERC20
+        });
+
+        AztecTypes.AztecAsset memory outputAssetA = AztecTypes.AztecAsset({
+            id: 4,
+            erc20Address: GUSD,
+            assetType: AztecTypes.AztecAssetType.ERC20
+        });
+
+        deal(WETH, address(bridge), swapAmount);
+
+        (uint256 outputValueA, , ) = bridge.convert(
+            inputAssetA,
+            emptyAsset,
+            outputAssetA,
+            emptyAsset,
+            swapAmount,
+            0,
+            encodedPath,
+            address(0)
+        );
+
+        IERC20(outputAssetA.erc20Address).transferFrom(address(bridge), rollupProcessor, outputValueA);
+
+        assertGt(outputValueA, 0);
+    }
+
+    function testSwapLowToHighDecimals(uint48 _swapAmount) public {
+        // Trying to swap anywhere from 1 USDC to 10 million USDC
+        uint256 swapAmount = bound(_swapAmount, 1e6, 1e13);
+        uint256 tokenInDecimals = 6;
+
+        uint256 quote = QUOTER.quoteExactInput(abi.encodePacked(USDC, uint24(500), WETH), swapAmount);
+        uint256 desiredMinPrice = quote / swapAmount;
```

Comment:
Why do you compute the `desiredMinPrice` as this? Say we swap 1000e6 USDC to 1e18 eth. Then we would a minPrice of 1e18/1e9 = 1e9. To me, it would seem more intuitive if the price is defined as the amount of asset B you receive for every whole asset A you supply. This is in practice also what you are doing when you are multiplying before doing the encoding.

Reply:
Desired minPrice is computed here this way so that the min amount out in convert(...) simply matches quote from the quoter. With this I test that the test doesn't revert with InsufficientAmountOut(...).